### PR TITLE
Remove Beats central management

### LINKED ISSUE #23908: Remove Beats Central management code
Beats central management has been marked as deprecated for a few releases and we want to remove it before 7.last.

Need to be synchronized with the Fleet team for https://github.com/elastic/kibana/issues/90614

### PR DESCRIPTION
**Pull request by michel-laterman**
## What does this PR do?

Remove the Kibana beats central management feature as it has been
replaced by Fleet. Remove xpack command injection and replace with
an includes mod. Restructure the xpack/libbeat/management module
to include all fleet code in the directory (no api or fleet subdirs).

## Checklist

- [x] My code follows the style guidelines of this project
- [x] ~~I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- [x] ~~I have made corresponding change to the default configuration files~~
- [x] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

1) Run ES 8 SNAPSHOT `elastic-package stack up --version=8.0.0-SNAPSHOT -v`
2) Package agent as a snapshot (on macos) `cd x-pack/elastic-agent && SNAPSHOT=true DEV=true PLATFORMS=darwin mage package`
3) Untar and enroll the agent using instructions from the Kibana fleet page
4) start agent and view in ES

## Related issues

- Closes #23908

### CODE REVIEW

**Comment by ph** [comment]
@jen-huang do you want to synchronize this removal with fleet or once we are green we can  merge it?

**Comment by michel-laterman** [comment]
jenkins run tests

**Comment by michel-laterman** [comment]
jenkins run tests

**Comment by michel-laterman** [comment]
package

**Comment by ph** [comment]
@michel-laterman Did you rebase with master? I see its the arm CI Job that are failling.

**Comment by jen-huang** [comment]
@ph @michel-laterman Merge at will. On Kibana side we have a draft PR for BCM removal https://github.com/elastic/kibana/pull/99789 already so it should follow closely regardless.

File: x-pack/libbeat/management/blacklist.go [hunk 1]
```diff
@@ -61,7 +60,7 @@ func NewConfigBlacklist(cfg ConfigBlacklistSettings) (*ConfigBlacklist, error) {
 }
 
 // Detect an error if any of the given config blocks is blacklisted
-func (c *ConfigBlacklist) Detect(configBlocks api.ConfigBlocks) Errors {
+func (c *ConfigBlacklist) Detect(configBlocks ConfigBlocks) Errors {
```
Comment (urso) [inline x-pack/libbeat/management/blacklist.go:63]: @ph @blakerouse Can we remove the filtering for output types (and others) from libbeat as well? These have been introduced in the context of Central Management, in order to disallow some potential risky setting.

Agent should disallow the `console` and `file` input? Filebeat modules should not be configurable anyways. For customizations we already have capabilities in Agent.
Do I miss something why we should keep 'blacklisting' functionality in libbeat?

File: x-pack/auditbeat/cmd/root.go [hunk 2]
```diff
@@ -23,5 +23,4 @@ func init() {
 	settings := auditbeatcmd.AuditbeatSettings()
 	settings.ElasticLicensed = true
 	RootCmd = auditbeatcmd.Initialize(settings)
-	xpackcmd.AddXPack(RootCmd, auditbeatcmd.Name)
```
Comment (urso) [inline x-pack/auditbeat/cmd/root.go:26]: Why was this line removed?

File: x-pack/libbeat/management/blacklist.go (hunk 1 above)
Comment (ph) [reply x-pack/libbeat/management/blacklist.go:63]: I agree we can remove it, but lets keep the PR focus on removing BCM we can do a follow-up to remove the "reject/accept" functionality after we have this merge. 

Note this old feature is now unnecessary to keep because of the `capabilities` of Elastic Agent.

File: x-pack/libbeat/management/config.go [hunk 3]
```diff
@@ -5,112 +5,40 @@
...
+	Type   string
+	Blocks []*ConfigBlock
+}
+
+// ConfigBlocks holds a list of type + configs objects
+type ConfigBlocks []ConfigBlocksWithType
+
 func defaultConfig() *Config {
 	return &Config{
-		Mode:   ModeCentralManagement,
```
Comment (michel-laterman) [inline x-pack/libbeat/management/config.go:108]: I don't think this should have many side effects, but it's worth noting that I changed the default mode (also not sure if we should define a `Mode` attribute and `ModeFleet` constant anymore

File: x-pack/libbeat/management/config.go [hunk 4]
```diff
@@ -119,30 +47,34 @@ func defaultConfig() *Config {
...
+		return nil, err
 	}
+	return &reload.ConfigWithMeta{
+		Config: config,
+	}, nil
+}
 
-	data, err := yaml.Marshal(tmp)
+// ConfigBlocksEqual returns true if the given config blocks are equal, false if not
+func ConfigBlocksEqual(a, b ConfigBlocks) (bool, error) {
```
Comment (michel-laterman) [inline x-pack/libbeat/management/config.go:67]: I don't think that this is used by fleet, should it also be removed?

File: x-pack/libbeat/management/error.go [hunk 5]
```diff
@@ -21,21 +19,13 @@ type ErrorType string
...
 	Err  error
 }
 
-// EventType returns a ErrorEvent.
-func (e *Error) EventType() api.EventType {
-	return ErrorEvent
-}
-
 // MarshalJSON transform an error into a JSON document.
 func (e *Error) MarshalJSON() ([]byte, error) {
```
Comment (michel-laterman) [inline x-pack/libbeat/management/error.go:30]: Are the MarshalJSON/UnmarshalJSON functions used by fleet?

Comment (urso) [reply x-pack/libbeat/management/error.go:30]: @blakerouse @michalpristas Any idea?

File: x-pack/libbeat/management/config.go (hunk 4 above)
Comment (urso) [reply x-pack/libbeat/management/config.go:67]: If it's not being used, it should be removed. CI/compiler should complain if you removed too much (I hope).

File: x-pack/libbeat/management/error.go (hunk 5 above)
Comment (blakerouse) [reply x-pack/libbeat/management/error.go:30]: I do not believe they are used.

File: x-pack/libbeat/management/config.go (hunk 3 above)
Comment (michel-laterman) [reply x-pack/libbeat/management/config.go:108]: removed

**Comment by urso** [comment]
Doc build failed with:

```
INFO:build_docs:Bad cross-document links:
INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/kibana/master/managing-beats.html contains broken links to:
INFO:build_docs:   - en/beats/filebeat/master/configuration-central-management.html
INFO:build_docs:   - en/beats/metricbeat/master/configuration-central-management.html
```

**Comment by michel-laterman** [comment]
Docs will be removed as part of the Kibana PR https://github.com/elastic/kibana/pull/99789

File: x-pack/libbeat/management/config.go (hunk 3 above)
Comment (michalpristas) [reply x-pack/libbeat/management/config.go:108]: for this removal please update `x-pack/elastic-agent/spec/*.yml` then `mage update` will need to be run.

File: x-pack/libbeat/management/error.go [hunk 6]
```diff
@@ -21,58 +17,13 @@ type ErrorType string
 // ConfigError is the type of error send when an unpack or a blacklist happen.
 var ConfigError = ErrorType("CONFIG")
 
-// ErrorEvent is the event type when an error happen.
-var ErrorEvent = api.EventType("ERROR")
-
 // Error is a config error to be reported to kibana.
 type Error struct {
```
Comment (michalpristas) [inline x-pack/libbeat/management/error.go:21]: i winder if we need to keep it here, we are coupling error to type and UUID but Error method just calls `Error` of nested one and we dont expose coupled information anywhere.

Comment (urso) [reply x-pack/libbeat/management/error.go:21]: Good point. It looks that the only reason for having this type was json marshalling/unmarshalling. Might indeed be obsolete as well.

File: x-pack/libbeat/management/config.go (hunk 3 above)
Comment (michel-laterman) [reply x-pack/libbeat/management/config.go:108]: done

**Comment by urso** [comment]
The packaging problems should be fixed in master by now.

**Review by michalpristas — APPROVED** [review]
code looks ok, i think all things were addressed. 
Tested ok, standalone and managed

**Review by urso — APPROVED** [review]
